src/Microsoft.AspNet.Mvc.Core/Infrastructure/DependencyContextAssemblyProvider.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Extensions.DependencyModel;
+
+namespace Microsoft.AspNet.Mvc.Infrastructure
+{
+    public class DependencyContextAssemblyProvider: IAssemblyProvider
+    {
+        /// <summary>
+        /// Gets the set of assembly names that are used as root for discovery of
+        /// MVC controllers, view components and views.
+        /// </summary>
+        // DefaultControllerTypeProvider uses CandidateAssemblies to determine if the base type of a POCO controller
+        // lives in an assembly that references MVC. CandidateAssemblies excludes all assemblies from the
+        // ReferenceAssemblies set. Consequently adding WebApiCompatShim to this set would cause the ApiController to
+        // fail this test.
+        protected virtual HashSet<string> ReferenceAssemblies { get; } = new HashSet<string>(StringComparer.Ordinal)
+        {
+            "Microsoft.AspNet.Mvc",
+            "Microsoft.AspNet.Mvc.Abstractions",
+            "Microsoft.AspNet.Mvc.ApiExplorer",
+            "Microsoft.AspNet.Mvc.Core",
+            "Microsoft.AspNet.Mvc.Cors",
+            "Microsoft.AspNet.Mvc.DataAnnotations",
+            "Microsoft.AspNet.Mvc.Formatters.Json",
+            "Microsoft.AspNet.Mvc.Formatters.Xml",
+            "Microsoft.AspNet.Mvc.Localization",
+            "Microsoft.AspNet.Mvc.Razor",
+            "Microsoft.AspNet.Mvc.Razor.Host",
+            "Microsoft.AspNet.Mvc.TagHelpers",
+            "Microsoft.AspNet.Mvc.ViewFeatures",
+            "Microsoft.AspNet.PageExecutionInstrumentation.Interfaces",
```

Comment:
Could you remove this? The package got removed but I missed removing it in the `DefaultAssemblyProvider`